tools/eslint-rules/restrict-let-in-for-loops.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @fileoverview Prohibit the use of `let` in `for` loops.
+ * @author Jessica Quynh Tran
+ */
+
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  create(context) {
+
+    const msg = 'Use of `let` declaration within a for-loop is ' +
+                'not recommended. Please use `var` instead.';
+
+    function testForLoop(node) {
+      return node.init && node.init.kind === 'let';
+    }
+
+    function testForInOfLoop(node) {
+      return node.left && node.left.kind === 'let';
+    }
+
+    /**
+     * Report function to test if the for-loop is declared using `let`.
+     */
+    function checkForLet(node) {
+      if (testForLoop(node) || testForInOfLoop(node)) {
+        context.report(node, msg);
+      }
+    }
+
+    return {
+      'ForStatement': (node) => checkForLet(node),
+      'ForInStatement': (node) => checkForLet(node),
```

Comment:
I think the suggestion from @princejwesley is a good one. The other tests that use a wrapper are also passing `context` because it is out of scope the way they are written, but not this one.